Cross Compile support with cmake build system

### Observed behavior
`smth modelcompiler: cannot execute binary file: Exec format error`

### Expected behavior
Having modelcompiler compiled for the build machines architecture

I worked around this problem by executing a `modelcompiler` that is installed on the build machine when cross compiling for now. Idk how you want this isssue to resolved.

```diff
--- CMakeLists.txt	2019-02-03 11:56:16.000000000 +0100
+++ -	2019-02-03 19:12:47.221852468 +0100
@@ -241,11 +241,19 @@
 
 # Optimize the models after the modelcompiler is built.
 # This really shouldn't be done inside the source tree...
+if(CMAKE_CROSSCOMPILING)
+add_custom_command(TARGET modelcompiler POST_BUILD
+	COMMAND ${CMAKE_COMMAND} -E env SDL_VIDEODRIVER=dummy modelcompiler -b inplace
+	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+	COMMENT "Optimizing models" VERBATIM
+)
+else()
 add_custom_command(TARGET modelcompiler POST_BUILD
 	COMMAND ${CMAKE_COMMAND} -E env SDL_VIDEODRIVER=dummy $<TARGET_FILE:modelcompiler> -b inplace
 	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
 	COMMENT "Optimizing models" VERBATIM
 )
+endif()
 
 install(TARGETS ${PROJECT_NAME} modelcompiler savegamedump
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
```

Currently when using cmake to Cross compile Pioneer for any system that can't executed on the build system (e.g. targeting ppc/arm while building on x86) will result in having a modelcompiler binary that can't be executed on the build machine, but the cmake script requires it.
I worked around this by just executing a previously compiled modelcompiler binary that I installed into $PATH.

This is what happens currently
```
[100%] Building CXX object CMakeFiles/pioneerLib.dir/src/posix/FileSystemPosix.cpp.o
[100%] Building CXX object CMakeFiles/pioneerLib.dir/src/posix/OSPosix.cpp.o
[100%] Linking CXX static library libpioneerLib.a
[100%] Built target pioneerLib
Scanning dependencies of target modelcompiler
Scanning dependencies of target pioneer
[100%] Building CXX object CMakeFiles/modelcompiler.dir/src/modelcompiler.cpp.o
[100%] Building CXX object CMakeFiles/pioneer.dir/src/main.cpp.o
[100%] Linking CXX executable modelcompiler
Optimizing models
/builddir/pioneer-20190203/build/modelcompiler: /builddir/pioneer-20190203/build/modelcompiler: cannot execute binary file
make[2]: *** [CMakeFiles/modelcompiler.dir/build.make:101: modelcompiler] Error 126
make[2]: *** Deleting file 'modelcompiler'
make[1]: *** [CMakeFiles/Makefile2:118: CMakeFiles/modelcompiler.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Linking CXX executable pioneer
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -239,13 +239,23 @@ set_target_properties(${PROJECT_NAME} modelcompiler savegamedump pioneerLib PROP
 	CXX_EXTENSIONS ON
 )
 
-# Optimize the models after the modelcompiler is built.
-# This really shouldn't be done inside the source tree...
-add_custom_command(TARGET modelcompiler POST_BUILD
-	COMMAND ${CMAKE_COMMAND} -E env SDL_VIDEODRIVER=dummy $<TARGET_FILE:modelcompiler> -b inplace
-	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-	COMMENT "Optimizing models" VERBATIM
-)
+if (CMAKE_CROSSCOMPILING)
+	find_program(MODELCOMPILER modelcompiler DOC "modelcompiler executable for the host")
+else (CMAKE_CROSSCOMPILING)
+	set(MODELCOMPILER $<TARGET_FILE:modelcompiler>)
+endif (CMAKE_CROSSCOMPILING)
+
+if (MODELCOMPILER)
+	# Optimize the models.
+	# This really shouldn't be done inside the source tree...
+	add_custom_command(TARGET modelcompiler POST_BUILD
+		COMMAND ${CMAKE_COMMAND} -E env SDL_VIDEODRIVER=dummy ${MODELCOMPILER} -b inplace
+		WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+		COMMENT "Optimizing models" VERBATIM
+	)
+else (MODELCOMPILER)
+	message(WARNING "No modelcompiler provided, models won't be optimized!")
+endif(MODELCOMPILER)
 
 install(TARGETS ${PROJECT_NAME} modelcompiler savegamedump
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}

